src/IntelliTect.Coalesce.CodeGeneration.Knockout/Generators/Views/TableView.cs

```diff
@@ -79,10 +83,12 @@ public override void BuildOutput(HtmlCodeBuilder b)
                         {
                             if (!prop.Type.IsCollection)
                             {
+                                string sortIconVisibleLogic = $"orderBy() != '{prop.Name}' &&  orderByDescending() != '{prop.Name}'";
                                 using (b.TagBlock("th", "sortable-header", dataBind: $"click: function(){{orderByToggle('{prop.Name}')}}"))
                                 {
-                                    b.Line(prop.DisplayName);
-                                    b.Line($"<i class=\"pull-right fa\" data-bind=\"css:{{'fa-caret-up': orderBy() == '{prop.Name}', 'fa-caret-down': orderByDescending() == '{prop.Name}'}}\"></i>");
+                                    b.Line($"<span>{prop.DisplayName}&nbsp;&nbsp;");
+                                    b.Line($@"<i class=""fa"" data-bind=""css:{{'fa-caret-up': orderBy() == '{prop.Name}', 'fa-caret-down': orderByDescending() == '{prop.Name}', 'fa-sort': {sortIconVisibleLogic} }}"" style =""float: right; padding: .3em 0 0 0 ""></i>");
```

Comment:
You could change this to use the string version of the `css` binding (which is also the same as the `class` binding). E.g. `css: orderBy() == '{prop.Name}' ? 'fa-caret-up' : orderByDescending() == '{prop.Name}' ? 'fa-caret-down' : 'fa-sort'`. This would eliminate the need to evaluate `orderBy` and `orderByDescending` two times each.

Reply:
Good catch.